src/test/java/nl/novi/serviceconnect/services/TransactionServiceTest.java

```diff
@@ -0,0 +1,230 @@
+
+        //Assert
+        assertEquals(1, outputDtoList.size());
+        assertEquals(1L, outputDtoList.get(0).id);
+    }
+
+    @Test
+    public void testGetAllTransaction_NoTransactionsFound() {
+
+        //Act
+        when(transactionRepository.findAll()).thenReturn(new ArrayList<>());
+
+        //Assert
+        assertThrows(RecordNotFoundException.class, () -> transactionService.getAllTransaction());
+    }
+
+    @Test
+    public void shouldGetTransactionById() {
+
+        //Arrange
+        Transaction mockTransaction = new Transaction();
+        mockTransaction.setId(1L);
+        mockTransaction.setInvoice("invoice.pdf");
+
+        //Act
+        when(transactionRepository.findById(1L)).thenReturn(Optional.of(mockTransaction));
+        ResponseEntity<byte[]> responseEntity = transactionService.getTransactionById(1L);
+
+        //Assert
+        assertNotNull(responseEntity);
+        assertNotNull(responseEntity.getBody());
+        assertEquals("invoice.pdf", responseEntity.getHeaders().getContentDisposition().getFilename());
+    }
+
+    @Test
+    public void testCreateTransaction_InvalidServiceRequestId() {
+
+        //Act
+        TransactionInputDto inputDto = new TransactionInputDto(new Date(2024-5-16),true,1L);
+
+        //Assert
+        assertThrows(RecordNotFoundException.class, () -> transactionService.createTransaction(inputDto));
+        verify(transactionRepository, never()).save(any(Transaction.class));
+    }
+
+    @Test
+    public void shouldSaveTransaction() {
+
+        //Arrange
+        Transaction mockTransaction = new Transaction();
+        mockTransaction.setId(1L);
+
+        //Act
+        transactionService.saveTransaction(mockTransaction);
+
+        //Assert
+        verify(transactionRepository, times(1)).save(mockTransaction);
+    }
+
+    @Test
+    public void whenNoTransactionFoundShouldReturn_TransactionNotFound() {
+
+        //Act
+        when(transactionRepository.findById(1L)).thenReturn(Optional.empty());
+
+        //Assert
+        assertThrows(RecordNotFoundException.class, () -> transactionService.getTransactionById(1L));
+    }
+
+    @Test
+    public void shouldUpdateTransaction() {
+
+        //Arrange
+        TransactionUpdateDto inputDto = new TransactionUpdateDto();
+        inputDto.setPayed(true);
+        inputDto.setTransactionDate(new Date());
+
+        Transaction existingTransaction = new Transaction();
+        existingTransaction.setId(1L);
+
+        //Act
+        when(transactionRepository.findById(1L)).thenReturn(Optional.of(existingTransaction));
+        TransactionOutputDto outputDto = transactionService.updateTransaction(1L, inputDto);
+
+        //Assert
+        assertNotNull(outputDto);
+        assertTrue(outputDto.isPayed);
+        assertNotNull(outputDto.transactionDate);
+    }
+
+    @Test
+    public void shouldValidateServiceRequestId_ServiceRequestIdExists() {
```

Comment:
Waarom zou hij een Exception gooien als de id bestaat?

Reply:
mag je geen transaction maken als het dezelfde service Request Id in database bestaat, en die method gooied exception als de id in de database heeft gevonden, dus de test is validering ServiceRequestId